Parse/src/main/java/com/parse/ParseRelation.java

```diff
@@ -193,7 +208,13 @@ public void remove(T object) {
   }
 
   /* package for tests */ ParseObject getParent() {
-    return parent;
+    if(parent == null){
+      return null;
+    }
+    if(parent.get() == null){
+      return ParseDecoder.get().decodePointer(parentClassName, parentObjectId);
```

Comment:
Let's just use `ParseObject.createWithoutData(String, String)` instead here so it'll be more concise.